content/posts/redis-modules-into-arm-land-part1.md

```diff
@@ -0,0 +1,275 @@
+
+By "workstation" I refer to a Linux or macOS host, that holds one's development environment and git repositories. As a side note,  I warmly recommend using a desktop PC (that's another blog post), though most people use laptops. In either case, I also recommend having some virtualization infrastructure on your workstation, VMware Workstation/Fusion or VirtualBox being good solutions.
+
+So, we need to establish a network connection between the RPi and the workstation. As mentioned before, RPi 4 has a 1Gbps NIC, which is a great improvement over the RPi 3 with its 100Mbps NIC. Connecting the RPi to a network is simple: get any gigabit Ethernet unmanaged switch and two Ethernet cables, then hook the RPi to the switch and connect the switch to your gateway Ethernet port. You can hook your workstation to the switch as well.
+
+At this stage, we need to gather some information from the workstation. 
+
+First, we need to determine UID & GID of the workstation user that owns the views:
+
+```
+id
+```
+
+We'll call them MY-UID & MY-GID.
+
+Next, we need to find out what's our time zone
+
+```
+timedatectl
+```
+
+We'll call it MY-TIMEZONE.
+
+Finally, we'll need your workstation's IP:
+
+```
+ip a
+```
+
+We'll call if MY-WORKSTATION-IP.
+
+#### RPi Configuration
+
+During the initial setup, you'll need to have the RPi connected to a monitor and a keyboard. Once setup is done, it can be controlled from your workstation via SSH.
+
+Another good practice is to avoid cloning source code into the RPi, but rather share it between your workstation and RPi via NFS.
+
+For convenience, I assume we operate as root (via ```sudo bash```, for instance).
+
+So let's get on with the configuration:
+
+##### Hostname
+
+```
+hostnamectl set-hostname MY-HOSTNAME
+```
+
+##### Time zone and NTP
+
+```
+# recall MY-TIMEZONE from previous section
+timedatectl set-timezone MY-TIMEZONE
+# confirm settings
+timedatectl
+```
+
+##### SSH server
+
+```
+apt-get -qq update
+apt install -qy openssh-server
+systemctl start openssh-server
+systemctl enable openssh-server
+```
+
+##### IP
+
+```
+ip a
+# record ip address, connect via ssh from workstation
+```
+
+##### NFS client
+
+```
+apt install -qy nfs-common
+mkdir -p /mnt/views
+ln -s /mnt/views /v
+```
+
+Add the following to ```/etc/hosts```:
+
+```
+workstation MY-WORKSTATION-IP 
+```
+
+Also, add the following line to ```/etc/fstab```:
+
+```
+workstation:/views /mnt/views nfs defaults 0 0
+```
+
+##### Utilities
+
+```
+apt install -qy tmux ca-certificates curl wget htop mc tmux
+```
+
+#### On the workstation side
+
+Regarding git repositories, I'll use the following terminology and structure throughout the discussion. This is not essential and one may organize matters differently. The term "view" refers to a set of git repository clones, that one uses in a certain context. For instance, if we work on RedisEdge modules in the context of ARM compilation, we'll end up with the following directory structure:

```

Comment:
Change this sentence to: "The term "view" refers to a set of git repository clones that one uses in a certain context."